org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/TaskViewContentProvider.java

```diff
@@ -71,9 +71,20 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
         // flatten the tree of Gradle projects to a list, similar
         // to how Eclipse projects look in the Eclipse Project explorer
         List<BaseProjectNode> allProjectNodes = Lists.newArrayList();
-        for (OmniEclipseProject project : projects) {
-            if (project.getParent() == null) {
-                collectProjectNodesRecursively(project, null, allProjectNodes);
+        
+        if(this.taskView.getState().isGroupProjects()) {
```

Comment:
I think we could use a better naming here. It should express that the project hierarchy is (or is not) flattened in the Tasks view. How about `TaskViewState.isProjectHierarchyFlattened()`?

Reply:
Yeah